### PR TITLE
Add POST/articles/:article_id/comments

### PR DESCRIPTION
**Pull request by gwdawson**

### CODE REVIEW

**Review by karlaferry — CHANGES_REQUESTED** [review]
Looks good! If you would please do the error handling and resubmit for approval that would be great. 😄

**Review by karlaferry — APPROVED** [review]
⬇️⬇️⬇️

**Review by karlaferry — CHANGES_REQUESTED** [review]
Actually, I don't see any error handling! Have you done any?

File: __tests__/articles.test.js [hunk 1]
```diff
@@ -48,6 +48,25 @@ describe('testing article endpoints', () => {
       expect(articles).toHaveLength(12);
     });
   });
+  describe('POST /api/articles/:article_id/comments', () => {
+    test('should return an object including keys {author, body, article_id, created_at, comment_id}', async () => {
+      const { body } = await request(app)
```
[the remark this answers is not in the document]
Comment (gwdawson) [reply __tests__/articles.test.js:54]: i like the suggestion but unfortunately prettier formats it so it look more confusing.

File: __tests__/articles.test.js [hunk 2]
```diff
@@ -48,6 +48,38 @@ describe('testing article endpoints', () => {
...
+    });
+    test('should return 400 when given an invalid username', async () => {
+      const { body } = await request(app)
+        .post('/api/articles/1/comments')
+        .send({ username: 'lurkerr', body: 'My username is lurker' })
+        .expect(400);
+      expect(body).toEqual(expect.objectContaining({ message: 'POST request must include a valid username' }));
+    });
+  });
+
```
Comment (jadelandeg) [inline __tests__/articles.test.js:82]: nice! Still need more error handling though, what could go wrong with the article id?

**Review by jadelandeg — CHANGES_REQUESTED** [review]
more error handling please

**Review by dentednerd — APPROVED** [review]
Great work Gareth!

**Pull request merged**